pandas/core/dtypes/dtypes.py

```diff
@@ -1665,17 +1665,6 @@ def _check_fill_value(self):
                     FutureWarning,
                     stacklevel=find_stack_level(),
                 )
-        elif isinstance(self.subtype, CategoricalDtype):
-            # TODO: is this even supported?  It is reached in
-            #  test_dtype_sparse_with_fill_value_not_present_in_data
-            if self.subtype.categories is None or val not in self.subtype.categories:
-                warnings.warn(
-                    "Allowing arbitrary scalar fill_value in SparseDtype is "
```

Comment:
Was this supposed to be removed?

Reply:
Yes. We determined that the isinstance on L1668 is impossible